### PR TITLE
dvc: 2.17.0 -> 2.43.1

### PR DESCRIPTION
**Pull request by mattmelling**
###### Description of changes

Upgrade and fix build for `dvc`, adding new dependencies `dvc-http`, `dvc-studio-client`, `iterative-telemetry`, updating `dvclive`, `dvc-task`.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>dvc</li>
    <li>dvc-with-remotes</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-studio-client</li>
    <li>python310Packages.dvc-task</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.iterative-telemetry</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-studio-client</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.iterative-telemetry</li>
  </ul>
</details>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dvclive/default.nix [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
 , ruamel-yaml
-, setuptools
+, setuptools-scm
 , tabulate
 }:
 
 buildPythonPackage rec {
   pname = "dvclive";
-  version = "1.3.0";
+  version = "1.2.2";
```
Comment (fabaff) [inline pkgs/development/python-modules/dvclive/default.nix:14]: Instead of downgrading the module the constraint of the consumer should be relaxed.

Comment (mattmelling) [reply pkgs/development/python-modules/dvclive/default.nix:14]: Oops, careless on my part. Internalized it as 1.2.2 -> 1.3.0 despite clearly doing the opposite :suspect: The `setuptools-scm` change is still required to provide the right version information though.

Comment (ryanswrt) [reply pkgs/development/python-modules/dvclive/default.nix]: DVC is /very/ sensitive to its package versions, relaxing constraints have broken things for alternative backends multiple times. Please consider including the exact versions they specify as much as possible.

**Comment by ryanswrt** [comment]
Tried this version and the `enableGoogle` option no longer works - it seems to require the [dvc-gs](https://github.com/iterative/dvc-gs) package now

File: pkgs/development/python-modules/dvclive/default.nix (hunk 1 above)
Comment (mattmelling) [reply pkgs/development/python-modules/dvclive/default.nix]: dvc specifies `dvclive>=1.2.2` so no need to relax, this was just dumb on my part.

**Comment by mattmelling** [comment]
Thanks @ryanswrt, I will test your PR on my fork

**Comment by mattmelling** [comment]
Added in @ryanswrt 's changes as well as additional packages for other cloud providers. Now waiting for the upgrade to `fsspec` requested in #208736

**Comment by NickCao** [comment]
The fsspec upgrade has been merged, are we good to go?

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 213038` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>dvc</li>
    <li>dvc-with-remotes</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-task</li>
    <li>python310Packages.gcsfs</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.gcsfs</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-studio-client</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.iterative-telemetry</li>
    <li>python310Packages.sshfs</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-studio-client</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.iterative-telemetry</li>
    <li>python311Packages.sshfs</li>
  </ul>
</details>

**Comment by mattmelling** [comment]
@NickCao It looks like the PR that updated fsspec has been updated to remove that change, hence the failures. I'll either add a new PR to upgrade that module or see if we can relax the deps on this one.

**Comment by fabaff** [comment]
I started to split out part of this PR to have smaller block to proceed.

- python3Packages.sshfs: #221138

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 213038` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>dvc-with-remotes</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.gcsfs</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-studio-client</li>
    <li>python311Packages.gcsfs</li>
    <li>python311Packages.scmrepo</li>
  </ul>
</details>
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>dvc</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-studio-client</li>
    <li>python310Packages.dvc-task</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.iterative-telemetry</li>
    <li>python310Packages.scmrepo</li>
    <li>python310Packages.sqltrie</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.iterative-telemetry</li>
    <li>python311Packages.sqltrie</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Depends on #221132

**Comment by bcdarwin** [comment]
Results of `nixpkgs-review` on x86_64-NixOS:

```
2 packages failed to build:
python311Packages.dvclive python311Packages.dvclive.dist

68 packages built:
dvc dvc-with-remotes dvc-with-remotes.dist dvc.dist python310Packages.dvc-azure python310Packages.dvc-azure.dist python310Packages.dvc-data python310Packages.dvc-data.dist python310Packages.dvc-gs python310Packages.dvc-gs.dist python310Packages.dvc-http python310Packages.dvc-http.dist python310Packages.dvc-objects python310Packages.dvc-objects.dist python310Packages.dvc-render python310Packages.dvc-render.dist python310Packages.dvc-s3 python310Packages.dvc-s3.dist python310Packages.dvc-ssh python310Packages.dvc-ssh.dist python310Packages.dvc-studio-client python310Packages.dvc-studio-client.dist python310Packages.dvc-task python310Packages.dvc-task.dist python310Packages.dvclive python310Packages.dvclive.dist python310Packages.hydra-core python310Packages.hydra-core.dist python310Packages.iterative-telemetry python310Packages.iterative-telemetry.dist python310Packages.nocturne python310Packages.nocturne.dist python310Packages.omegaconf python310Packages.omegaconf.dist python310Packages.pydevd python310Packages.pydevd.dist python310Packages.scmrepo python310Packages.scmrepo.dist python311Packages.dvc-azure python311Packages.dvc-azure.dist python311Packages.dvc-data python311Packages.dvc-data.dist python311Packages.dvc-gs python311Packages.dvc-gs.dist python311Packages.dvc-http python311Packages.dvc-http.dist python311Packages.dvc-objects python311Packages.dvc-objects.dist python311Packages.dvc-render python311Packages.dvc-render.dist python311Packages.dvc-s3 python311Packages.dvc-s3.dist python311Packages.dvc-ssh python311Packages.dvc-ssh.dist python311Packages.dvc-studio-client python311Packages.dvc-studio-client.dist python311Packages.hydra-core python311Packages.hydra-core.dist python311Packages.iterative-telemetry python311Packages.iterative-telemetry.dist python311Packages.nocturne python311Packages.nocturne.dist python311Packages.omegaconf python311Packages.omegaconf.dist python311Packages.pydevd python311Packages.pydevd.dist python311Packages.scmrepo python311Packages.scmrepo.dist
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 213038` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.dvclive</li>
    <li>python311Packages.dvclive.dist</li>
  </ul>
</details>
<details>
  <summary>68 packages built:</summary>
  <ul>
    <li>dvc</li>
    <li>dvc-with-remotes</li>
    <li>dvc-with-remotes.dist</li>
    <li>dvc.dist</li>
    <li>python310Packages.dvc-azure</li>
    <li>python310Packages.dvc-azure.dist</li>
    <li>python310Packages.dvc-data</li>
    <li>python310Packages.dvc-data.dist</li>
    <li>python310Packages.dvc-gs</li>
    <li>python310Packages.dvc-gs.dist</li>
    <li>python310Packages.dvc-http</li>
    <li>python310Packages.dvc-http.dist</li>
    <li>python310Packages.dvc-objects</li>
    <li>python310Packages.dvc-objects.dist</li>
    <li>python310Packages.dvc-render</li>
    <li>python310Packages.dvc-render.dist</li>
    <li>python310Packages.dvc-s3</li>
    <li>python310Packages.dvc-s3.dist</li>
    <li>python310Packages.dvc-ssh</li>
    <li>python310Packages.dvc-ssh.dist</li>
    <li>python310Packages.dvc-studio-client</li>
    <li>python310Packages.dvc-studio-client.dist</li>
    <li>python310Packages.dvc-task</li>
    <li>python310Packages.dvc-task.dist</li>
    <li>python310Packages.dvclive</li>
    <li>python310Packages.dvclive.dist</li>
    <li>python310Packages.hydra-core</li>
    <li>python310Packages.hydra-core.dist</li>
    <li>python310Packages.iterative-telemetry</li>
    <li>python310Packages.iterative-telemetry.dist</li>
    <li>python310Packages.nocturne</li>
    <li>python310Packages.nocturne.dist</li>
    <li>python310Packages.omegaconf</li>
    <li>python310Packages.omegaconf.dist</li>
    <li>python310Packages.pydevd</li>
    <li>python310Packages.pydevd.dist</li>
    <li>python310Packages.scmrepo</li>
    <li>python310Packages.scmrepo.dist</li>
    <li>python311Packages.dvc-azure</li>
    <li>python311Packages.dvc-azure.dist</li>
    <li>python311Packages.dvc-data</li>
    <li>python311Packages.dvc-data.dist</li>
    <li>python311Packages.dvc-gs</li>
    <li>python311Packages.dvc-gs.dist</li>
    <li>python311Packages.dvc-http</li>
    <li>python311Packages.dvc-http.dist</li>
    <li>python311Packages.dvc-objects</li>
    <li>python311Packages.dvc-objects.dist</li>
    <li>python311Packages.dvc-render</li>
    <li>python311Packages.dvc-render.dist</li>
    <li>python311Packages.dvc-s3</li>
    <li>python311Packages.dvc-s3.dist</li>
    <li>python311Packages.dvc-ssh</li>
    <li>python311Packages.dvc-ssh.dist</li>
    <li>python311Packages.dvc-studio-client</li>
    <li>python311Packages.dvc-studio-client.dist</li>
    <li>python311Packages.hydra-core</li>
    <li>python311Packages.hydra-core.dist</li>
    <li>python311Packages.iterative-telemetry</li>
    <li>python311Packages.iterative-telemetry.dist</li>
    <li>python311Packages.nocturne</li>
    <li>python311Packages.nocturne.dist</li>
    <li>python311Packages.omegaconf</li>
    <li>python311Packages.omegaconf.dist</li>
    <li>python311Packages.pydevd</li>
    <li>python311Packages.pydevd.dist</li>
    <li>python311Packages.scmrepo</li>
    <li>python311Packages.scmrepo.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
I will deal with `dvclive` later today.

**Review by fabaff — APPROVED** [review]

**Pull request merged**